src/Api/Request/RequestBuilder.php

```diff
@@ -80,57 +82,36 @@ public function __construct(
     /**
      * @param HttpRequest $httpRequest
      * @param ExternalSearch $externalSearch
-     * @param int|null $category
-     * @return bool|Request
+     * @param Category|null $category
+     * @return array
      */
     public function build(HttpRequest $httpRequest, ExternalSearch $externalSearch, $category = null)
     {
-        $request = $this->createRequestObject();
-        $request = $this->setDefaultValues($request, $this->getRequestType($httpRequest, $category));
-        $request = $this->parametersBuilder->setSearchParams($request, $httpRequest, $externalSearch, $category);
+        $this->request['requestType'] = $this->getRequestType($httpRequest, $category);
+        $this->setDefaultValues($this->getRequestType($httpRequest, $category));
+        $this->request = array_merge($this->request, $this->parametersBuilder->setSearchParams($httpRequest, $externalSearch, $category));
 
-        return $request;
+        return $this->request;
     }
 
     /**
-     * @return bool|Request
+     * @return array
      */
     public function buildAliveRequest()
     {
-        $request = $this->createRequestObject();
-        $request->setConfiguration(Plugin::API_CONFIGURATION_KEY_TIME_OUT, 1);
-        $request->setUrl(
-            $this->getUrl(self::ALIVE_REQUEST_TYPE)
-        )->setParam('shopkey', $this->pluginConfig->getShopKey());
-
-        return $request;
-    }
-
-    /**
-     * @return Request
-     */
-    public function createRequestObject()
-    {
-        return pluginApp(Request::class);
+        $this->request['shopUrl'] = $this->getUrl(self::ALIVE_REQUEST_TYPE);
```

Comment:
The argument was removed

```suggestion
        $this->request['shopUrl'] = $this->getUrl();
```